src/Interval.cpp

```diff
@@ -157,5 +157,108 @@ Expr Interval::neg_inf_noinline() {
     return Interval::neg_inf_expr;
 }
 
+ConstantInterval::ConstantInterval() = default;
+
+ConstantInterval::ConstantInterval(int64_t min, int64_t max)
+    : min(min), max(max), min_defined(true), max_defined(true) {
+    internal_assert(min <= max);
+}
+
+ConstantInterval ConstantInterval::everything() {
+    return ConstantInterval();
+}
+
+ConstantInterval ConstantInterval::single_point(int64_t x) {
+    return ConstantInterval(x, x);
+}
+
+ConstantInterval ConstantInterval::bounded_below(int64_t min) {
+    ConstantInterval result(min, min);
+    result.max_defined = false;
+    return result;
+}
+
+ConstantInterval ConstantInterval::bounded_above(int64_t max) {
+    ConstantInterval result(max, max);
+    result.min_defined = false;
+    return result;
+}
+
+bool ConstantInterval::is_everything() const {
+    return !min_defined && !max_defined;
+}
+
+bool ConstantInterval::is_single_point() const {
+    return !is_everything() && min == max;
+}
+
+bool ConstantInterval::is_single_point(int64_t x) const {
+    return !is_everything() && min == x && max == x;
+}
+
+bool ConstantInterval::has_upper_bound() const {
+    return max_defined;
+}
+
+bool ConstantInterval::has_lower_bound() const {
+    return min_defined;
+}
+
+bool ConstantInterval::is_bounded() const {
+    return !is_everything();
```

Comment:
This definition is not the same as `Interval`'s definition, which is `is_bounded -> has_upper_bound() && has_lower_bound()`, which could be confusing. However, I don't think I've found a case where this method is called for a `ConstantInterval`, so maybe it could just be removed